modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/Flamingos2.scala

```diff
@@ -85,18 +86,27 @@ object Flamingos2 {
     case _                    => WindowCover.OPEN
   }
 
+  implicit def grismFromSPDisperser(d: Disperser): Grism = d match {
+    case Disperser.NONE    => Grism.Open
+    case Disperser.R1200HK => Grism.R1200HK
+    case Disperser.R1200JH => Grism.R1200JH
+    case Disperser.R3000   => Grism.R3000
+  }
+
+  private def disperserFromObserveType(observeType: String, d: Disperser): Grism = observeType match {
+    case DARK_OBSERVE_TYPE    => Grism.Dark
+    case _                    => d
+  }
+
   def ccConfigFromSequenceConfig(config: Config): TrySeq[CCConfig] = ( for {
+      obsType <- config.extract(OBSERVE_KEY / OBSERVE_TYPE_PROP).as[String]
       // WINDOW_COVER_PROP is optional. If not present, then window cover position is inferred from observe type.
-      p <- config.extract(INSTRUMENT_KEY / WINDOW_COVER_PROP).as[WindowCover] match {
-            case a: \/-[WindowCover] => a
-            case _         => config.extract(OBSERVE_KEY / OBSERVE_TYPE_PROP).as[String]
-                              .map(windowCoverFromObserveType)
-          }
+      p <- config.extract(INSTRUMENT_KEY / WINDOW_COVER_PROP).as[WindowCover].recover(PartialFunction((x:ConfigUtilOps.ExtractFailure) => windowCoverFromObserveType(obsType)))
```

Comment:
So we try to read it from the sequence but if not we try to guess it from the observation type?